ush/fv3gfs_downstream_nems.sh

```diff
@@ -1,294 +1,188 @@
-#-Fanglin Yang, March 2017
-#  1. Modified for FV3GFS, using NCEP-NCO standard output name convention
-#  2. Add option24 to turn on bitmap in grib2 file (from Wen Meng)
-#-Wen Meng, January 2018, add flag PGB1F for turning on/ogg grib1 pgb data at 1.00 deg. generation.
-#-Wen Meng, Feburary 2018
-#  1. Add flag PGBS for turning on/off pgb data at 1.0 and 0.5 deg. generation frequency of FHOUT_PGB defined.
-#-Wen Meng, October 2019
-#  1. Use bilinear interpolation for LAND. It can trancate land-sea mask as 1 or 0.
-#-Wen Meng, November 2019
-#  1. Modify sea icea cover via land-sea mask.
-#-----------------------------------------------------------------------
+source "${HOMEgfs}/ush/preamble.sh" "${FH}"
 
-source "$HOMEgfs/ush/preamble.sh" "$FH"
-
-export downset=${downset:-1}
-export DATA=${DATA:-/ptmpd2/$LOGNAME/test}
-export CNVGRIB=${CNVGRIB:-${grib_util_ROOT}/bin/cnvgrib}
-export COPYGB2=${COPYGB2:-${grib_util_ROOT}/bin/copygb}
+# Programs used
 export WGRIB2=${WGRIB2:-${wgrib2_ROOT}/bin/wgrib2}
-export GRBINDEX=${GRBINDEX:-${wgrib2_ROOT}/bin/grbindex}
-export RUN=${RUN:-"gfs"}
-export cycn=$(echo $CDATE |cut -c 9-10)
-export TCYC=${TCYC:-".t${cycn}z."}
-export PREFIX=${PREFIX:-${RUN}${TCYC}}
-export PGB1F=${PGB1F:-"NO"}
-export FHOUT_PGB=${FHOUT_PGB:-3}
-export PGBS=${PGBS:-"NO"} #YES-- generate 1.00 and 0.50 deg pgb data
-export MODICEC=${MODICEC:-$USHgfs/mod_icec.sh}
-
-#--wgrib2 regrid parameters
-export option1=' -set_grib_type same -new_grid_winds earth '
-export option21=' -new_grid_interpolation bilinear  -if '
-export option22=":(CRAIN|CICEP|CFRZR|CSNOW|ICSEV):"
-export option23=' -new_grid_interpolation neighbor -fi '
-export option24=' -set_bitmap 1 -set_grib_max_bits 16 -if '
-export option25=":(APCP|ACPCP|PRATE|CPRAT):"
-export option26=' -set_grib_max_bits 25 -fi -if '
-export option27=":(APCP|ACPCP|PRATE|CPRAT|DZDT):"
-export option28=' -new_grid_interpolation budget -fi '
-export grid0p25="latlon 0:1440:0.25 90:721:-0.25"
-export grid0p5="latlon 0:720:0.5 90:361:-0.5"
-export grid1p0="latlon 0:360:1.0 90:181:-1.0"
-export grid2p5="latlon 0:144:2.5 90:73:-2.5"
-
-unset paramlist paramlistb
-if [ $FH -eq -1 ] ; then
-  #export paramlist=/global/save/Hui-Ya.Chuang/gfs_trunk/sib/fix/global_1x1_paramlist_g2.anl
-  export paramlist=${paramlist:-$PARMpost/global_1x1_paramlist_g2.anl}
-  export paramlistb=${paramlistb:-$PARMpost/global_master-catchup_parmlist_g2}
-  export fhr3=anl
-  export PGBS=YES
-elif [ $FH -eq 0 ] ; then
-  export paramlist=${paramlist:-$PARMpost/global_1x1_paramlist_g2.f000}
-  export paramlistb=${paramlistb:-$PARMpost/global_master-catchup_parmlist_g2}
-  export fhr3=000
-  export PGBS=YES
+CNVGRIB=${CNVGRIB:-${grib_util_ROOT}/bin/cnvgrib}
+GRBINDEX=${GRBINDEX:-${wgrib2_ROOT}/bin/grbindex}
+
+# Scripts used
+GFSDWNSH=${GFSDWNSH:-"${HOMEgfs}/ush/fv3gfs_dwn_nems.sh"}
+
+# variables used here and in $GFSDWNSH
+PGBOUT2=${PGBOUT2:-"master.grib2"}  # grib2 file from UPP
+FH=${FH:-0}  # Forecast hour to process
```

Comment:
Let's head problems off at the pass and make sure `FH` doesn't have leading zeros:
```suggestion
FH=$(( ${FH:-0} ))  # Forecast hour to process
```